### PR TITLE
Resolve validation issues in endgame

### PR DESCRIPTION
**Pull request by Flanker32**
What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 1]
```diff
@@ -56,12 +59,41 @@ public DockerHostCreationDialog(final Project project) {
...
         txtCertPath.addValueChangedListener(ignore -> resetValidationMessage());
+        txtCertPath.addValidator(this::validateCertPath);
         txtCertPath.addActionListener(new ComponentWithBrowseButton.BrowseFolderActionListener<>("Select Cert for Docker Host", null, txtCertPath,
                 project, FileChooserDescriptorFactory.createSingleFolderDescriptor(), TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT));
     }
 
+    private AzureValidationInfo validateCertPath() {
+        final String value = txtCertPath.getValue();
+        if (StringUtils.isNotEmpty(value) && !FileUtil.exists(value)) {
+            return AzureValidationInfo.error(String.format("File %s does not exists", value), txtCertPath);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java:76]:
```suggestion
            return AzureValidationInfo.error(AzureString.format("File %s does not exists", value), txtCertPath);
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 2]
```diff
@@ -56,12 +59,41 @@ public DockerHostCreationDialog(final Project project) {
...
+        }
+        return AzureValidationInfo.success(txtCertPath);
+    }
+
+    private AzureValidationInfo validateDockerHost() {
+        final String value = txtDockerHost.getValue();
+        try {
+            final URI uri = URI.create(value);
+            if (StringUtils.isBlank(uri.getScheme())) {
+                return AzureValidationInfo.error("Invalid host value, schema of uri could not be empty", txtDockerHost);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java]:
```suggestion
                return AzureValidationInfo.error("Invalid Docker host URI, schema of uri could not be empty", txtDockerHost);
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 3]
```diff
@@ -56,12 +59,41 @@ public DockerHostCreationDialog(final Project project) {
...
+            }
+            if (StringUtils.isBlank(uri.getHost())) {
+                return AzureValidationInfo.error("Invalid host value, host of uri could not be empty", txtDockerHost);
+            }
+        } catch (final Exception e) {
+            return AzureValidationInfo.error(String.format("Invalid host value, %s", e.getMessage()), txtDockerHost);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java]:
```suggestion
            return AzureValidationInfo.error(String.format("Invalid Docker host URI, %s", e.getMessage()), txtDockerHost);
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 4]
```diff
@@ -56,12 +59,41 @@ public DockerHostCreationDialog(final Project project) {
...
+
+    private AzureValidationInfo validateDockerHost() {
+        final String value = txtDockerHost.getValue();
+        try {
+            final URI uri = URI.create(value);
+            if (StringUtils.isBlank(uri.getScheme())) {
+                return AzureValidationInfo.error("Invalid host value, schema of uri could not be empty", txtDockerHost);
+            }
+            if (StringUtils.isBlank(uri.getHost())) {
+                return AzureValidationInfo.error("Invalid host value, host of uri could not be empty", txtDockerHost);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java]:
```suggestion
                return AzureValidationInfo.error("Invalid Docker host URI, host of uri could not be empty", txtDockerHost);
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 5]
```diff
@@ -18,6 +19,7 @@
 import com.microsoft.azure.toolkit.intellij.container.model.DockerHost;
 import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java:20]:
```suggestion
import com.microsoft.azure.toolkit.lib.common.bundle.AzureString;
import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webapponlinux/WebAppOnLinuxDeployConfiguration.java [hunk 6]
```diff
@@ -85,6 +86,10 @@ public void validate() throws ConfigurationException {
         checkAzurePreconditions();
         validateDockerHostConfiguration(getDockerHostConfiguration());
         validateDockerImageConfiguration(getDockerImageConfiguration());
+        // registry
+        if (StringUtils.isEmpty(getContainerRegistryId())) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webapponlinux/WebAppOnLinuxDeployConfiguration.java:90]: do we set `required` on the corresponding component?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 7]
```diff
@@ -56,12 +60,41 @@ public DockerHostCreationDialog(final Project project) {
...
         txtCertPath.addValueChangedListener(ignore -> resetValidationMessage());
+        txtCertPath.addValidator(this::validateCertPath);
         txtCertPath.addActionListener(new ComponentWithBrowseButton.BrowseFolderActionListener<>("Select Cert for Docker Host", null, txtCertPath,
                 project, FileChooserDescriptorFactory.createSingleFolderDescriptor(), TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT));
     }
 
+    private AzureValidationInfo validateCertPath() {
+        final String value = txtCertPath.getValue();
+        if (StringUtils.isNotEmpty(value) && !FileUtil.exists(value)) {
+            return AzureValidationInfo.error(AzureString.format("File %s does not exists", value), txtCertPath);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java]:
```suggestion
            return AzureValidationInfo.error(String.format("File %s does not exists", value), txtCertPath);
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java [hunk 8]
```diff
@@ -10,14 +10,17 @@
...
 import com.intellij.openapi.ui.ComponentWithBrowseButton;
 import com.intellij.openapi.ui.TextComponentAccessor;
+import com.intellij.openapi.util.io.FileUtil;
 import com.intellij.ui.AnimatedIcon;
 import com.microsoft.azure.toolkit.intellij.common.AzureDialog;
 import com.microsoft.azure.toolkit.intellij.common.AzureTextInput;
 import com.microsoft.azure.toolkit.intellij.common.component.AzureFileInput;
 import com.microsoft.azure.toolkit.intellij.container.AzureDockerClient;
 import com.microsoft.azure.toolkit.intellij.container.model.DockerHost;
+import com.microsoft.azure.toolkit.lib.common.bundle.AzureString;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-containerregistry/src/main/java/com/microsoft/azure/toolkit/intellij/containerregistry/component/DockerHostCreationDialog.java]:
```suggestion
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webapponlinux/WebAppOnLinuxDeployConfiguration.java (hunk 6 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webapponlinux/WebAppOnLinuxDeployConfiguration.java:90]: yes, they have been set as required

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**